Refactoring browser and manager params into dataclasses (#807)
mozilla/OpenWPM: docs/Configuration.md

# Browser and Platform Configuration <!-- omit in toc -->

The browser and platform can be configured by two separate classes. The
platform configuration options can be set in `manager_params`, while the
browser configuration options can be set in `browser_params`. The default
settings are given in `openwpm/config.py` in class `BrowserParams` and `ManagerParams`

> To load the default configuration parameter just makes instances of `config::ManagerParams` and `config::BrowserPramas`. For example:

Reviewer: It's generally considered bad writing to use words like "just" because it makes the reader feel like they should have known this and demotivates them.
Please remove it from this sentence.

Reviewer: ```suggestion
To load the default configuration create instances of `config::ManagerParams` and `config::BrowserParams`. For example:
```

Author: Good to know, committed!...

```python
from openwpm.config import BrowserParams, ManagerParams
num_browsers=5

> manager_params = ManagerParams()

Reviewer: Please set the num_browsers in the ManagerParams()

* `log_directory`
  * The directory in which to output platform logs. The
    directory given will be created if it does not exist.

> * `log_file` -> suppored file extensions are `.log`

Reviewer: ```suggestion
* `log_file` -> supported file extensions are `.log`
```

  * The name of the log file to be written to `log_directory`.

> * `database_name` -> suppored file extensions are `.db`, `.sqlite`

Reviewer: ```suggestion
* `database_name` -> supported file extensions are `.db`, `.sqlite`
```

  * The following options are supported:
    * `always`: Accept all third-party cookies
    * `never`: Never accept any third-party cookies
    * `from_visited`: Only accept third-party cookies from sites that have been visited as a first party.
* `donottrack`
  * Set to `True` to enable Do Not Track in the browser.

> * `tracking_protection`

Reviewer: Good job on finding the typo
